src/DataPersister/ChainDataPersister.php

```diff
@@ -51,7 +51,7 @@ public function persist($data)
     {
         foreach ($this->persisters as $persister) {
             if ($persister->supports($data)) {
-                $persister->persist($data);
+                return $persister->persist($data) ?? $data;
```

Comment:
I know it's a bit late, but wouldn't this prevent _all_ persisters to run, and wouldn't this stop at the first persister that supports this data ? what if I need to apply multiple data persisters ?

Reply:
Good catch... We may return after the loop.instead of inside. WYT? (ping @gorghoa)

Reply:
indeed, nice catch…

> We may return after the loop instead of inside. WYT?

But from which persister shall we return? First one? Last one?  (somehow, this question is in fact valid for the former implementation. The last was the winner, but was it “safe” to assume it was the one that should win?).

 Maybe the best option in term of confidence should be not return data from persister but somehow re-trigger a read item operation. At the cost of an extra read query.

Reply:
> afterthoughts

It was exactly what i've in mind (last win)

Reply:
https://github.com/api-platform/core/pull/2064